extensions/wikia/Email/Controller/WatchedPageController.class.php

```diff
@@ -26,8 +33,11 @@ public function initEmail() {
 
 		$this->title = \Title::newFromText( $titleText, $titleNamespace );
 		$this->summary = $this->getVal( 'summary' );
-		$this->currentRevId = $this->getVal( 'currentRevId' );
-		$this->previousRevId = $this->getVal( 'previousRevId' );
+
+		if ( $this->title instanceof \Title ) {
+			$this->currentRevId = $this->title->getLatestRevID( \Title::GAID_FOR_UPDATE );
+			$this->previousRevId = $this->title->getPreviousRevisionID( $this->currentRevId, \Title::GAID_FOR_UPDATE );
+		}
```

Comment:
What does this change do?  ```EmailNotifcation``` can send us the current and previous revision IDs without having to do a DB lookup against the master.  Also, I could see there being a race condition if two edits come in at once.